app/src/main/java/com/github/se/travelpouch/model/activity/ActivityViewModel.kt

```diff
@@ -0,0 +1,128 @@
+package com.github.se.travelpouch.model.activity
+
+import android.content.Context
+import android.util.Log
+import android.widget.Toast
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import com.github.se.travelpouch.model.Location
+import com.google.firebase.Firebase
+import com.google.firebase.Timestamp
+import com.google.firebase.firestore.firestore
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+
+class ActivityModelView(val activityRepositoryFirebase: ActivityRepository) : ViewModel() {
+
+  // create factory
+  companion object {
+    val Factory: ViewModelProvider.Factory =
+        object : ViewModelProvider.Factory {
+          @Suppress("UNCHECKED_CAST")
+          override fun <T : ViewModel> create(modelClass: Class<T>): T {
+            return ActivityModelView(ActivityRepositoryFirebase(Firebase.firestore)) as T
+          }
+        }
+  }
+
+  private val activities_ = MutableStateFlow<List<Activity>>(emptyList())
+  val activities: StateFlow<List<Activity>> = activities_.asStateFlow()
+
+  private val selectedActivity_ =
+      MutableStateFlow<Activity>(
+          Activity(
+              "uid",
+              "title",
+              "description",
+              Location(0.0, 0.0, Timestamp(0, 0), "location"),
+              Timestamp(0, 0),
+              null))
+  val selectedActivity: StateFlow<Activity> = selectedActivity_.asStateFlow()
+
+  private val onFailureTag = "ActivityViewModel"
+
+  /** This is the initialisation function of the model view */
+  fun init() {
+    activityRepositoryFirebase.init { getAllActivities() }
+  }
+
+  /** This function gets all the activities from the database */
+  fun getAllActivities() {
+    activityRepositoryFirebase.getAllActivities(
+        onSuccess = { activities_.value = it },
+        onFailure = { Log.e(onFailureTag, "Failed to get all activities", it) })
+  }
+
+  /**
+   * This function adds an activity to the database
+   *
+   * @param activity (Activity) : the activity to add to the database
+   */
+  fun addActivity(activity: Activity, context: Context) {
+    activityRepositoryFirebase.addActivity(
+        activity,
+        {
+          getAllActivities()
```

Comment:
for consistency and readability, wouldn't it be better to always use (onFailure = and onSuccess =) ?

Reply:
Yes, you are right